### PR TITLE
Refactor rendering process

### PR DESCRIPTION
**Pull request by timstallmann**
This is based on @plopesc PR but fixes a few bugs and removes the custom niceJson() method.

Refactor rendering process by:

- Moving code in views_geojson file to the GeoJson class to improve encapsulation and have all the code together. Functions there are called only from that class.
- Refactoring row generation, removing foreach loops done in every row, doing a previous preprocess of the data and reusing results of that preprocess. I tested with a feed with 500 results and its faster.
- Use `json_encode` with the `JSON_PRETTY_PRINT` flag to render previews in views directly

### CODE REVIEW

File: src/Plugin/views/style/GeoJson.php [hunk 1]
```diff
@@ -411,9 +407,9 @@ protected function renderFields(ResultRow $row, array $excluded_fields) {
     // - Views "excluded" fields.
     foreach (array_keys($this->view->field) as $id) {
       $field = $this->view->field[$id];
-      if (!in_array($id, $excluded_fields, TRUE) && !($field->options['exclude'])) {
-        // Allows you to customize the name of the property by setting a label to
-        // the field.
+      if (!in_array($id, $this->getExcludedFields(), TRUE) && !($field->options['exclude'])) {
```
Comment (plopesc) [inline src/Plugin/views/style/GeoJson.php]: @timstallmann I think this change will hit performance. You're calculating the excluded rows in every row process. and the result is the same always. That's why it was passed as a parameter to the method.

File: src/Plugin/views/style/GeoJson.php [hunk 2]
```diff
@@ -314,18 +314,19 @@ public function render() {
     }
     unset($this->view->row_index);
 
-    // Render the collection to JSON.
-    $json = Json::encode($features);
+    if (!empty($this->view->live_preview)) {
```
Comment (plopesc) [inline src/Plugin/views/style/GeoJson.php]: @timstallmann  doing this here you're not going to show the jsonp config in case of the view preview

Comment (timstallmann) [inline src/Plugin/views/style/GeoJson.php]: @plopesc I moved the jsonp line after this one, so it should catch, right?

File: src/Plugin/views/style/GeoJson.php (hunk 1 above)
Comment (timstallmann) [inline src/Plugin/views/style/GeoJson.php]: Oh good call! How about caching that within getExcludedFields itself then.. I can make that change

File: src/Plugin/views/style/GeoJson.php (hunk 2 above)
Comment (plopesc) [inline src/Plugin/views/style/GeoJson.php]: @timstallmann sorry,you're completely right

File: src/Plugin/views/style/GeoJson.php (hunk 1 above)
Comment (plopesc) [inline src/Plugin/views/style/GeoJson.php]: @timstallmann sounds like a plan :+1:

**Comment by timstallmann** [comment]
@plopesc just re-implemented the caching, realized we could just use the `$this->excludedFields` variable anyway since that was defined but being unused.

**Comment by plopesc** [comment]
:clap: :clap:

**Pull request merged**